### DISCUSSION TITLE
How to change minimum password length for self hosted

### DISCUSSION THREAD

**Discussion by christianfl** [supabase/supabase#7181] [Questions] [answered]
Hello!

I recently installed Supabase as a self hosted version in Docker. I want to set the minimum password length but cannot find how to do that in the docs.

I guess it should be a key in the .env file like the other things:

```
## General
SITE_URL=https://URL
ADDITIONAL_REDIRECT_URLS=
JWT_EXPIRY=3600
DISABLE_SIGNUP=false
```

Can someone tell me how to do that?

Thats in advance,
Christian

**Comment by dshukertjr** [comment, marked as answer]
Hi @christianfl!

You should be able to configure minimum password length with `GOTRUE_PASSWORD_MIN_LENGTH` environment variable. I found it on the [readme of gotrue](https://github.com/supabase/gotrue). 
```
GOTRUE_PASSWORD_MIN_LENGTH - int
Minimum password length, defaults to 6.
```

**Reply by christianfl** [reply]
Thank you very much!

**Comment by OyugoObonyo** [comment]
Can you reset the minimum password length on the managed supabase service or is this option only available in self-hosted Supabase?

**Reply by dshukertjr** [reply]
@OyugoObonyo 
You can find the minimum password length option in auth -> providers -> email in your Supabase dashboard for all Supabase projeects!
<img width="1382" alt="Screenshot 2023-03-31 at 11 46 06" src="https://user-images.githubusercontent.com/18113850/229010097-4dcfe250-a32f-4377-a142-112e8bca71c0.png">

**Reply by OyugoObonyo** [reply]
Great. I tried lowering the number to 4 to suit my 4-digit-PIN authentication strategy but the insisted minimum is still 6. How then can I go about implementing sign-in with a phone number and 4-digit PIN with Supabase? Would it be ideal for me to ask for a 4-digit PIN from users and then pad the pin in the backend with, let's say symbols so that they meet the 6-digit requirement?

**Reply by dshukertjr** [reply]
@OyugoObonyo 
We would suggest to stick with the minimum requirement that we have for security reasons. If you would like to take a less secure way to implement your auth feature, it would completely at your own responsibility.
